core/modules/node/node.module

```diff
@@ -2772,37 +2772,90 @@ function node_node_access($node, $op, $account) {
  *
  * @param $type
- *   The machine-readable name of the node type.
+ *   The machine-readable name of the content type.
+ *
+ *   If $type is empty or omitted, the default content-related permissions are
+ *   returned. These are "placeholder" permissions - the actual permissions
+ *   include the content type name in their name. For example, if the content
+ *   type is called "test", the actual permission for creating content of this
+ *   type will be "create test content" instead of just "create content" (which
+ *   is the respective placeholder permission). This is useful when the content
+ *   type name has not been created yet (for instance, when creating a new
+ *   content type in node_type_form_permissions().
  *
  * @return array
  *   An array of permission names and descriptions.
+ *
+ * @since 1.25.0 $type is optional.
  */
-function node_list_permissions($type) {
-  $info = node_type_get_type($type);
-
-  // Build standard list of node permissions for this type.
-  $perms = array(
-    "create $type content" => array(
-      'title' => t('%type_name: Create new content', array('%type_name' => $info->name)),
+function node_list_permissions($type = '') {
+  // Default, "placeholder" permissions when creating a new content type (the
+  // content type name has not been created yet).
+  $permissions = array(
+    'create content' => array(
+      'title' => t('Create new content'),
     ),
-    "edit own $type content" => array(
-      'title' => t('%type_name: Edit own content', array('%type_name' => $info->name)),
+    'edit own content' => array(
+      'title' => t('Edit own content'),
     ),
-    "edit any $type content" => array(
-      'title' => t('%type_name: Edit any content', array('%type_name' => $info->name)),
+    'edit any content' => array(
+      'title' => t('Edit any content'),
     ),
-    "delete own $type content" => array(
-      'title' => t('%type_name: Delete own content', array('%type_name' => $info->name)),
+    'delete own content' => array(
+      'title' => t('Delete own content'),
     ),
-    "delete any $type content" => array(
-      'title' => t('%type_name: Delete any content', array('%type_name' => $info->name)),
+    'delete any content' => array(
+      'title' => t('Delete any content'),
     ),
   );
 
-  return $perms;
+  // Existing content type. Generate the list of actual permissions, by adding
+  // the content type name in the names and descriptions of the placeholder
+  // permissions.
+  if (!empty($type) && $info = node_type_get_type($type)) {
+    foreach ($permissions as $placeholder_permission => $placeholder_permission_info) {
+      // Inject the content type name in the new permission name.
+      $actual_permission = node_generate_permission_from_placeholder($placeholder_permission, $type);
+      // Add the content type name as a prefix to the permission description.
+      $actual_permission_title = $info->name . ': ' . $placeholder_permission_info['title'];
```

Comment:
This line assumes that in every language the permission title can be `[content type]: [permission]`, but in right-to-left languages that should probably be `[permission] :[content type]`. It is also probable that languages that do not use a Latin script, the equivalent of `:` is a different character.

Reply:
Yeah, we should probably do this properly using `t()` and placeholders, rather than just concatenating here.

Reply:
Placeholders should probably give more context to translators, just in the case a language uses a different order when `!name` is a proper noun.